compiler/src/lntoamm/UserFunction.ts

```diff
@@ -33,92 +33,116 @@ class UserFunction implements Fn {
       if (statements[i].isReturnStatement()) {
         // There are unreachable statements after this line, abort
         throw new Error(`Unreachable code in function '${name}' after:
-${statements[i].statementAst.getText().trim()} on line ${statements[i].statementAst.start.line}:${statements[i].statementAst.start.column}`)
+${statements[i].statementAst.t.trim()} on line ${statements[i].statementAst.line}:${statements[i].statementAst.char}`)
       }
     }
     this.statements = statements
     this.pure = pure
   }
 
-  static fromAst(functionishAst: any, scope: Scope) { // TODO: Eliminate ANTLR
-    if (functionishAst instanceof LnParser.BlocklikesContext) {
-      if (functionishAst.functions() != null) {
-        return UserFunction.fromFunctionsAst(functionishAst.functions(), scope)
+  static fromAst(functionishAst: LPNode, scope: Scope) {
+    if (
+      functionishAst.has('fnname') ||
+      functionishAst.has('functions') ||
+      functionishAst.has('functionbody')
+    ) { // It's a `blocklike` node
+      if (functionishAst.has('functions')) {
+        return UserFunction.fromFunctionsAst(functionishAst.get('functions'), scope)
+      }
+      if (functionishAst.has('functionbody')) {
+        return UserFunction.fromFunctionbodyAst(functionishAst.get('functionbody'), scope)
       }
-      if (functionishAst.functionbody() != null) {
-        return UserFunction.fromFunctionbodyAst(functionishAst.functionbody(), scope)
+      if (functionishAst.has('fnname')) {
+        // TODO: We didn't cover this path before?
```

Comment:
does this need to get implemented right now?

Reply:
Well, can it be a follow-up? We apparently never ran into it before. I need to see what paths even lead to here, though. This is creating a function from the AST, so maybe it would be something as simple as `const myFn: function = someOtherFunction;`?